pkg/users/local.js

```diff
@@ -55,12 +57,12 @@ function AccountsPage() {
 
     const [details, setDetails] = useState(null);
     useEffect(() => {
-        getLogins().then(setDetails);
+        getLogins(shadow).then(setDetails);
 
         // Watch `/var/run/utmp` to register when user logs in or out
         const handle = cockpit.file("/var/run/utmp", { superuser: "try", binary: true });
         handle.watch(() => {
-            getLogins().then(setDetails);
+            getLogins(shadow).then(setDetails);
         });
         return handle.close;
     }, [path, accounts, shadow]);
```

Comment:
Unrelated to  the PR, but this makes at least loading the page call logins ~ three times. I know understand that my solution of dropping the dependencies is why the tests fail. I do wonder if we can do better, maybe checking if shadow changed?

Reply:
Doesn't this also leak watches? Each invocation of `handle.watch()` installs a new one, i.e. each time the effect runs. And nothing ever calls .close() on that. It would really be better to pull the entire watch code  out of the effect and into some useInit(), so that it runs *exactly* once.

Reply:
Please move the watch to useInit(). That also has a cleanup action where you can .close() the watch again.